src/ui/utils/resumeOperations.ts

```diff
@@ -0,0 +1,142 @@
+import {
+  ExecutionPoint,
+  PauseDescription,
+  SourceId,
+  SourceLocation,
+  loadedRegions,
+} from "@replayio/protocol";
+
+import { ThreadFront } from "protocol/thread";
+import { locationsInclude } from "protocol/utils";
+import { ReplayClientInterface } from "shared/client/types";
+
+export interface ResumeOperationParams {
+  point?: ExecutionPoint;
+  loadedRegions: loadedRegions;
+  sourceId?: SourceId;
+  locationsToSkip?: SourceLocation[];
+}
+
+type FindTargetCommand =
+  | "findRewindTarget"
+  | "findResumeTarget"
+  | "findStepInTarget"
+  | "findStepOutTarget"
+  | "findStepOverTarget"
+  | "findReverseStepOverTarget";
+
+async function findResumeTarget(
+  client: ReplayClientInterface,
+  findTargetCommand: FindTargetCommand,
+  {
+    point,
+    loadedRegions,
+    sourceId,
+    locationsToSkip,
+  }: ResumeOperationParams & { point: ExecutionPoint }
+) {
+  if (
+    loadedRegions.loaded.every(
+      region =>
+        BigInt(point) < BigInt(region.begin.point) || BigInt(point) > BigInt(region.end.point)
+    )
+  ) {
```

Comment:
nit: We have utility methods for this:

```suggestion
  if (!isRangeInRegions(region.begin, loadedRegions.loaded)) {
```

Reply:
Seems worth making this change?